hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AInputStream.java

```diff
@@ -1043,15 +1024,11 @@ private void readSingleRange(FileRange range, ByteBuffer buffer) {
    * @return result s3 object.
    * @throws IOException exception if any.
    */
-  private S3Object getS3ObjectAndValidateNotNull(final String operationName,
-                                                 final long position,
-                                                 final int length) throws IOException {
+  private ResponseInputStream<GetObjectResponse> getS3ObjectInputStream(
+      final String operationName, final long position, final int length) throws IOException {
     checkIfVectoredIOStopped();
-    S3Object objectRange = getS3Object(operationName, position, length);
-    if (objectRange.getObjectContent() == null) {
-      throw new PathIOException(uri,
-              "Null IO stream received during " + operationName);
-    }
+    ResponseInputStream<GetObjectResponse> objectRange =
```

Comment:
Since this returns the input stream directly, the additional null check is no longer needed.

Reply:
Consider whether to move the `checkIfVectoredIOStopped()` calls into `getS3Object()` and remove this method.

Reply:
Yes, I think we can refactor this.

Reply:
Also think about this https://issues.apache.org/jira/browse/HADOOP-17338 . An old related issue.

Reply:
Thanks @mukund-thakur, we don't expect the GetObject being gc'd issue to resurface as the getObject call now returns an input stream directly.